tests/Shared/TaskExtensions.cs

```diff
@@ -0,0 +1,135 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.InternalTesting;
+
+internal static class TaskExtensions
+{
+#if DEBUG
+    // Shorter duration when running tests with debug.
+    // Less time waiting for hang unit tests to fail in aspnetcore solution.
+    public const int DefaultTimeoutDuration = 5 * 1000;
+#else
+    public const int DefaultTimeoutDuration = 30 * 1000;
+#endif
+
+    public static TimeSpan DefaultTimeoutTimeSpan { get; } = TimeSpan.FromMilliseconds(DefaultTimeoutDuration);
+
+    public static Task DefaultTimeout(this Task task, int milliseconds = DefaultTimeoutDuration, [CallerFilePath] string? filePath = null, [CallerLineNumber] int lineNumber = default)
+    {
+        return task.TimeoutAfter(TimeSpan.FromMilliseconds(milliseconds), filePath, lineNumber);
+    }
+
+    public static Task DefaultTimeout(this Task task, TimeSpan timeout, [CallerFilePath] string? filePath = null, [CallerLineNumber] int lineNumber = default)
+    {
+        return task.TimeoutAfter(timeout, filePath, lineNumber);
+    }
+
+    public static Task DefaultTimeout(this ValueTask task, int milliseconds = DefaultTimeoutDuration, [CallerFilePath] string? filePath = null, [CallerLineNumber] int lineNumber = default)
+    {
+        return task.AsTask().TimeoutAfter(TimeSpan.FromMilliseconds(milliseconds), filePath, lineNumber);
+    }
+
+    public static Task DefaultTimeout(this ValueTask task, TimeSpan timeout, [CallerFilePath] string? filePath = null, [CallerLineNumber] int lineNumber = default)
+    {
+        return task.AsTask().TimeoutAfter(timeout, filePath, lineNumber);
+    }
+
+    public static Task<T> DefaultTimeout<T>(this Task<T> task, int milliseconds = DefaultTimeoutDuration, [CallerFilePath] string? filePath = null, [CallerLineNumber] int lineNumber = default)
+    {
+        return task.TimeoutAfter(TimeSpan.FromMilliseconds(milliseconds), filePath, lineNumber);
+    }
+
+    public static Task<T> DefaultTimeout<T>(this Task<T> task, TimeSpan timeout, [CallerFilePath] string? filePath = null, [CallerLineNumber] int lineNumber = default)
+    {
+        return task.TimeoutAfter(timeout, filePath, lineNumber);
+    }
+
+    public static Task<T> DefaultTimeout<T>(this ValueTask<T> task, int milliseconds = DefaultTimeoutDuration, [CallerFilePath] string? filePath = null, [CallerLineNumber] int lineNumber = default)
+    {
+        return task.AsTask().TimeoutAfter(TimeSpan.FromMilliseconds(milliseconds), filePath, lineNumber);
+    }
+
+    public static Task<T> DefaultTimeout<T>(this ValueTask<T> task, TimeSpan timeout, [CallerFilePath] string? filePath = null, [CallerLineNumber] int lineNumber = default)
+    {
+        return task.AsTask().TimeoutAfter(timeout, filePath, lineNumber);
+    }
+
+    [SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Required to maintain compatibility")]
```

Comment:
1. Do we have public analyzers turned on for our tests? That seems wrong.
2. What does `"Required to maintain compatibility"` mean for our tests? I don't think we need to worry about compatibility.

Reply:
It’s copied from dotnet/aspnetcore

Reply:
It would be good to add that in a comment on the file